crates/bevy_asset/src/lib.rs

```diff
@@ -232,7 +233,10 @@ impl Plugin for AssetPlugin {
     label = "invalid `Asset`",
     note = "consider annotating `{Self}` with `#[derive(Asset)]`"
 )]
-pub trait Asset: VisitAssetDependencies + TypePath + Send + Sync + 'static {}
+pub trait Asset: VisitAssetDependencies + TypePath + Send + Sync + 'static {
+    /// Register component hooks for [`Handle<Self>`].
```

Comment:
Can you say more about why this might be useful in the docs here?

Reply:
That would be more docs than the actual `register_component_hooks` function on `Component` :)

I'm not actually sure what to put here, I want to use it for code organization with `MaterialMeshBundle` but not sure that's the best thing to put here.